hamocc/mo_inventory_bgc.F90

```diff
@@ -462,15 +465,40 @@ subroutine inventory_bgc(kpie,kpje,kpke,dlxp,dlyp,ddpo,omask,iogrp)
       totaloxy = totaloxy + zocetratot(iano2)+zpowtratot(ipown2o)*0.5+zpowtratot(ipowno2)
     endif
 
+    if (use_river2omip) then
+      totalcarbon = totalcarbon + zocetratot(itdoc_lc)*rcar_tdoclc        &
+                                + zocetratot(itdoc_hc)*rcar_tdoclc
+      totalnitr   = totalnitr   + zocetratot(itdoc_lc)*rnit_tdoclc        &
+                                + zocetratot(itdoc_hc)*rnit_tdoclc
+      totaloxy   = totaloxy     + zocetratot(itdoc_lc)*roxy_tdoclc        &
+                                + zocetratot(itdoc_hc)*roxy_tdoclc
+      totalphos  = totalphos + zocetratot(itdoc_lc) + zocetratot(itdoc_hc)
+    endif
+
     if (do_rivinpt) then
-      totalcarbon = totalcarbon- (srivflux(irdoc)+srivflux(irdet))*rcar                            &
-           &                   - (srivflux(iralk)+srivflux(irdin)+srivflux(irdip))   ! =sco212
-      totalnitr   = totalnitr  - (srivflux(irdoc)+srivflux(irdet))*rnit - srivflux(irdin)
-      totalphos   = totalphos  - (srivflux(irdoc)+srivflux(irdet)+srivflux(irdip))
+      if (use_river2omip) then
+        totalcarbon = totalcarbon- srivflux(irdoc)*rcar-srivflux(irtdoc)*rcar_tdochc               &
+             &                   - srivflux(irdet)*rcar_tdoclc                                     &
+             &                   - srivflux(iralk)-srivflux(irdin)-srivflux(irdip)   ! =sco212
```

Comment:
Compare to what you apply in `mo_apply_rivin`

Reply:
Re-check - I believe after your changes in `mo_apply_rivin.F90`, the terms: `-srivflux(irdin)-srivflux(irdip)` are likely too much.